aws/data_source_aws_guardduty_detector_test.go

```diff
@@ -0,0 +1,74 @@
+package aws
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+func TestAccAWSGuarddutyDetectorDataSource_basic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:                  func() { testAccPreCheck(t) },
+		Providers:                 testAccProviders,
+		PreventPostDestroyRefresh: true,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsGuarddutyDetectorBasicResourceConfig(),
+				Check:  resource.ComposeTestCheckFunc(),
+			},
+			{
+				Config: testAccAwsGuarddutyDetectorBasicResourceDataConfig(),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrPair("data.aws_guardduty_detector.test", "id", "aws_guardduty_detector.test", "id"),
+					resource.TestCheckResourceAttrPair("data.aws_guardduty_detector.test", "status", "aws_guardduty_detector.test", "status"),
```

Comment:
The `status` attribute does not exist on the existing resource, so this needs to be updated to a regular `TestCheckResourceAttr()`

```
--- FAIL: TestAccAWSGuarddutyDetectorDataSource_basic (17.64s)
    testing.go:635: Step 1 error: Check failed: Check 2/4 error: data.aws_guardduty_detector.test: Attribute "status" is "ENABLED", but "status" is not set in aws_guardduty_detector.test
```